x-pack/plugins/upgrade_assistant/public/application/components/overview/identify_step/external_links.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FunctionComponent, useState, useEffect } from 'react';
+
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiLink, EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiPanel, EuiText } from '@elastic/eui';
+
+import { useKibana, DataPublicPluginStart } from '../../../../shared_imports';
+import {
+  DEPRECATION_LOGS_INDEX_PATTERN,
+  DEPRECATION_LOGS_SOURCE_ID,
+} from '../../../../../common/constants';
+
+const getDeprecationIndexPatternId = async (dataService: DataPublicPluginStart) => {
+  const { indexPatterns: indexPatternService } = dataService;
+
+  const results = await indexPatternService.find(DEPRECATION_LOGS_INDEX_PATTERN);
```

Comment:
The line that throws the error is the `indexPatternService.createAndSave`, we need to do so because it's required to have an index pattern if we want to link to the Discover app. The problem is that the index is not created until a log is emitted, which means if you enable the toggle for the first time (with a fresh es instance for example), the index wont exists and this error will pop up. 

We are probably going to have to add a check to see if the index exists before creating the index pattern, and we might also have to rely on this to determine if the setting is enabled or not. Perhaps in this scenario we can keep the toggle enabled and show some sort of loading state saying "as soon as a deprecation log exists you will be able to inspect them" kind of thing.